pkg/apimachinery/registered/registered.go

```diff
@@ -158,6 +162,44 @@ func IsRegistered(group string) bool {
 	return found
 }
 
+// IsRegisteredVersion returns if a version is registered.
+func IsRegisteredVersion(v unversioned.GroupVersion) bool {
+	_, found := registeredVersions[v]
+	return found
+}
+
+// IsThirdPartyAPIGroupVersion returns true if the api version is a user-registered group/version.
+func IsThirdPartyAPIGroupVersion(gv unversioned.GroupVersion) bool {
+	for ix := range thirdPartyGroupVersions {
+		if thirdPartyGroupVersions[ix] == gv {
+			return true
+		}
+	}
+	return false
+}
+
+// SetThirdPartyAPIGroupVersions sets the list of third party versions, overwrites anything existing.
+func SetThirdPartyAPIGroupVersions(gvs ...unversioned.GroupVersion) {
```

Comment:
This operates more like an `AddThirdPartyAPIGroupVersions`.  It's not stomping, just appending.

Seems like you should return a list groupVersions that were skipped, so callers can decide if they care.